### PR TITLE
Shadowling Updates

### PR DESCRIPTION
**Pull request by TheDZD**
Ports tgstation/-tg-station#10254, tgstation/-tg-station#10871, and tgstation/-tg-station#9927.
 - Fixes a few issues present in 10871.

### Functionality and Gameplay Changes
- Fixes floodlights not being extinguished by the shadowling Veil spell.
- Removes loop that goes through world looking for APCs. We need to implement an apc_list if we want to go through APCs in a sane way to overload all of the station's lightning when a shadowling ascends. Currently just overloads APCs in a large range ([might not be ideal either](https://github.com/ParadiseSS13/Paradise/pull/1603/files#diff-fa0219b3f784928d4fb20088f3968623R151)).
- Shadowling Ascendants get a transmit ability that allows them to send messages directly to world.
- Replaced shadowling ascendants' Glacial Blast with a Lightning Storm ability, which fires bolts of red lightning at all nearby mobs that are not shadowlings or thralls.
- Makes it so mobs with environment_smash >= 2 can smash *all* walls, not just reinforced ones. (There was a stupid istype check that for some reason made it so only reinforced walls could be smashed.)
- Does not port the surgical procedure to deconvert thralls, because Bay surgery is a lot more fickle than /tg/ surgery.
 - Also, did not port the change that prevents badmins from using shadowling spells (Still not a particularly good idea to use spells like Ascend outside of the gamemode/as a non-shadowling).
- Gives Shadowling Ascendants universal_speak, meaning everything can understand them.
- Shadowling ascendants are no longer harmed by explosions or Lord Singuloth.
- Ups shadowling ascendant melee damage to 60 from 35.
 - Also ups their environment_smash to 3, meaning they smash walls in one hit instead of taking several hits.
- Makes number of starting shadowlings scale up to 10% of the player count.
- Shadowling thralls are identifiable if you examine them and they're not wearing a mask, but they also get two spells.
 - Lesser Glare - Stuns (does not make them fall to the ground) a target and mutes them for a very short period of time.
 - Guise - Makes the Thrall difficult to see for a few seconds, not usable in brightly-lit areas.
- Renames Flash Freeze to Icy Veins
- The Annihilate and Hypnosis spells (Ascendant spells that gib and enthrall at a range respectively) now have no cooldown.
- Enthrall spell has no cooldown, still takes 30 seconds to enthrall somebody, can still be interrupted.
- Drain Thralls (spell acquired at 5 thralls) replaced with Drain Life, causes minor stamina and oxyloss damage to nearby humans and heals the shadowling for each target.
- Spacial Relocate (spell acquired at 9 thralls) replaced with Black Recuperation, a 5 minute cooldown spell that allows shadowlings to revive a thrall.
- Shadowlings are no longer named shadowling # after hatching, and now have a set of shadowling names.

### Coder Stuff

- Adds a new turf helper proc, get_lumcount(), which is used to determine the total brightness of a turf based on all of its lighting overlays.
- Shadowling Hatch and Ascend abilities are now spells, not mob procs.
- Bunch of spanclass changes to shadowling text.
- Refactors some of how Veil extinguishes mob and item lights.
- Adds the mind proc remove_spell(), takes a single path argument, take a guess at what it does.
- Slightly changes colors of shadowling span class.

### CODE REVIEW

**Comment by TheDZD** [comment]
How does this already have merge conflicts?

**Comment by Fox-McCloud** [comment]
once this has been updated with the more recent  changes Xhuis has made and tested, I'd be glad to try my hand at implementing the dethralling surgery.

**Comment by Fox-McCloud** [comment]
Looks fine to me.

File: code/game/gamemodes/shadowling/ascendant_shadowling.dm [hunk 1]
```diff
@@ -13,21 +14,29 @@
...
 	faction = list("faithless")
 
 /mob/living/simple_animal/ascendant_shadowling/Process_Spacemove(var/movement_dir = 0)
 	return 1 //copypasta from carp code
+
+/mob/living/simple_animal/ascendant_shadowling/ex_act(severity)
+	return //You think an ascendant can be hurt by bombs? HA
+
+/mob/living/simple_animal/ascendant_shadowling/singularity_act()
+	return 0 //Well hi, fellow god! How are you today?
```
Comment (tigercat2000) [inline code/game/gamemodes/shadowling/ascendant_shadowling.dm]: Heresy.

**Pull request merged**